### PR TITLE
SE - Non-nullable value types: Indexer and Range support

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Part of #6105

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs [hunk 1]
```diff
@@ -394,6 +394,34 @@ public void Main<T>() where T : new()
...
 
+#if NET
+
+        [TestMethod]
+        public void Index_SetsNotNull()
+        {
+            const string code = """
+            var index = ^0;
+            Tag("Index", index);
+            """;
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs:405]: +4
```suggestion
                var index = ^0;
                Tag("Index", index);
                """;
```

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs [hunk 2]
```diff
@@ -394,6 +394,34 @@ public void Main<T>() where T : new()
...
+            validator.ValidateTag("Index", x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+        }
+
+        [TestMethod]
+        public void Range_SetsNotNull()
+        {
+            const string code = """
+            var range = 0..1;
+            Tag("Range", range);
+            """;
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs:417]: +4
```suggestion
                var range = 0..1;
                Tag("Range", range);
                """;
```

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Comment by martin-strecker-sonarsource** [comment]
@pavel-mikula-sonarsource please merge.

**Pull request merged**